src/api/functional_tests.js

```diff
@@ -1,306 +1,272 @@
-import axios from 'axios'
-import moment from 'moment';
-
-async function _getSingleWorkflow(workflowId) {
-    const singleWorkflowRunApi = `https://api.github.com/repos/mozilla-mobile/mozilla-vpn-client/actions/workflows/${workflowId}/runs?branch=main`    
-    let return_data = [];
-
-    try {
-        const { data } = await axios.get(singleWorkflowRunApi)
-        for(let i=0; i < 12; i++){
-            const workflow = data.workflow_runs[i]
-            if(workflow.conclusion !== 'cancelled'){
-                const workflow_run = {
-                    id: workflow.id,
-                    workflow_id: workflow.workflow_id,
-                    check_suite_id: workflow.check_suite_id,
-                    name: workflow.name,
-                    started: workflow.run_started_at,
-                    title: workflow.display_title,
-                    status: workflow.status,
-                    conclusion: workflow.conclusion,
-                    branch: workflow.head_branch,
-                    actor_email: workflow.head_commit.author.email,
-                    actor_name: workflow.head_commit.author.name,
-                    actor_message: workflow.head_commit.message,
-                    actor_commit_time: workflow.head_commit.timestamp,
-                    sha: workflow.sha,
-                    event: workflow.event,
-                    node_id: workflow.node_id,
-                    url: workflow.url,
-                    html_url: workflow.html_url,
-                    created: workflow.created_at,
-                    updated: workflow.updated_at,
-                    run_attempt: workflow.run_attempt,
-                    referenced_wf: workflow.referenced_workflows[0],
-                    triggering_actor: workflow.triggering_actor.login,
-                    jobs_url: workflow.jobs_url,
-                    logs_url: workflow.logs_url,
-                    workflow_url: workflow.workflow_url            
-                }
-    
-                return_data.push(workflow_run)
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import axios from "axios";
+import moment from "moment";
 
-            }
-        }        
-    } catch(err){
-        console.log('Unable to fetch getSingleWorkflow', err);
-        return
+async function _getSingleWorkflow(workflowId) {
+  const singleWorkflowRunApi = `https://api.github.com/repos/mozilla-mobile/mozilla-vpn-client/actions/workflows/${workflowId}/runs?branch=main`;
+  let return_data = [];
+
+  try {
+    const { data } = await axios.get(singleWorkflowRunApi);
+    for (let i = 0; i < 12; i++) {
```

Comment:
more of a rate limiting caution, will increase once i've addressed the limits

Reply:
can you point me to the documentation of this limit? It sounds strange that a simple wait fixes the issue. Plus, this should be done in pre-processing via a github action and not when the page is loaded. What do you think? Otherwise the loading of this page will take forever..